src/users/users.service.ts

```
@@ -1,24 +1,58 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { User } from './user';
+
+
 
-// This should be a real class/interface representing a user entity
-export type User = any;
 
 @Injectable()
-export class UsersService {
-  private readonly users = [
-    {
-      userId: 1,
-      username: 'Mustafa',
-      password: 'darsh',
-    },
-    {
-      userId: 2,
-      username: 'Ali',
-      password: 'march',
-    },
-  ];
-
-  async findOne(username: string): Promise<User | undefined> {
-    return this.users.find(user => user.username === username);
-  }
+export class UserService {
+    constructor(@InjectRepository(User) private userRepository: Repository<User>){}
+
+    async createUser(item){
+        const {email} = item
+        const user = await this.userRepository.findOneBy({email})       
+        if (user) {
+            return {message: "already exists"}
```

remind me to talk about error codes.
Also, you can just throw an exception